internal/pkg/template/override/rule.go

```diff
@@ -3,15 +3,50 @@
 
 package override
 
+import "gopkg.in/yaml.v3"
+
+const (
```

Comment:
good call!

Reply:
Added more comments. Usually I don't write a lot of comments to compensate the poor readability of the code itself but this could be an exception since it is complicated and I can't think of a way to make it more readable 🥺